types/jest/index.d.ts

```diff
@@ -756,19 +787,28 @@ declare namespace jest {
         /**
          * Used when you want to check that an item is in a list.
          * For testing the items in the list, this uses `===`, a strict equality check.
+         *
+         * Optionally, you can provide a type for the expected value via a generic.
+         * This is particuarly useful for ensuring expected objects have the right structure.
          */
-        toContain(expected: any): R;
+        toContain<E>(expected: E): R;
```

Comment:
```suggestion
        toContain<E = any>(expected: E): R;
```

Explicit fallback rather than implicit